src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpAutomaticBraceCompletion.cs

```diff
@@ -469,7 +469,10 @@ class B : A
 }
 ");
 
-            VisualStudio.Editor.SendKeys("override Goo(");
+            VisualStudio.Editor.SendKeys("override ");
+            Assert.True(VisualStudio.Editor.IsCompletionActive());
```

Comment:
The failure here is a true product regression. The IDE is no longer waiting for completion to appear after a completion trigger is typed, so for users who type fast (including the integration test machines), completion will sometimes complete things and sometimes will not.

Reply:
This is a change in editor behavior, right? Is there some other way our tests should indicate to wait, or is this the right approach? @AmadeusW

Reply:
Import completion is turned on via experiment for all 16.5 previews, it might be involved during integration tests here. I have made some perf improvement in 16.5p2, so if "responsive completion" is indeed causing this, hopefully it's less of a problem after switching to 16.5p2.

Reply:
These integration tests are running on 16.5 Preview 2

Reply:
Perhaps this will help:
```
[Import] IAsyncCompletionBroker broker;

// Arrange
var sessionComputedEvent = new AutoResetEvent(false);
Broker.CompletionTriggered += (s, triggerArgs) => 
{ 
    triggerArgs.CompletionSession.ItemsUpdated += (o, e) => sessionComputedEvent.Set(); 
    // you may also use triggerArgs.CompletionSession.Dismissed event which also it fires after commit
};

// Act ...
await sessionComputedEvent.ToTask(MaxTestDelay);
// Act ...

```

Reply:
@AmadeusW Do you know how to turn it off in test?

Reply:
@genlu This is unrelated to the **Responsive code completion** feature. I have that feature disabled and IntelliCode still breaks the IDE behavior locally.

Reply:
>  The IntelliCode package reduces it all the way to 0.

Is this a known bug? we should disable IntelliCode until it's fixed then

Reply:
I'm not sure if it's a known bug. I believe the best resolution for the integration tests is to disable IntelliCode as part of the test initialization here:
https://github.com/dotnet/roslyn/blob/deafbee29842c3dcd56a8c3b040fbcd2d9fe97b8/src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs#L327-L344

Reply:
@dpoeschl Could you check with IntelliCode team about this? Also do you know how to disable it in integration test?

Reply:
@sharwell 

> IntelliCode still breaks the IDE behavior locally
> The IntelliCode package reduces it all the way to 0.

Sorry, what? They are "just" a Roslyn completion API extension. They provide no items while their model is initially loading (without blocking), and then provide items once the model is loaded. Is this the behavior you're talking about?

Reply:
For disabling IntelliCode during test initialization, paging @vivlimmsft for any ideas there.

Reply:
I think the easiest way to accomplish this is to set the `(privateregistry.bin)\Software\Microsoft\VisualStudio\(current instance)\ApplicationPrivateSettings\Microsoft\VisualStudio\IntelliCode\CSharpBaseModel` to `1*System.Int32*2`, similarly to how roaming settings & background download UI are disabled for the test. This will set the IntelliCode tri-state setting for C# completions to 'Disabled'.

The change would probably look like this, though I haven't tried this line of code:
```csharp
 Process.Start(vsRegEditExeFile, $"set \"{installationPath}\" {Settings.Default.VsRootSuffix} HKCU \"ApplicationPrivateSettings\\Microsoft\\VisualStudio\\IntelliCode\" CSharpBaseModel string \"1*System.Int32*2\"").WaitForExit(); 
```

Reply:
@vivlimmsft note that the interfering feature here is not the IntelliCode completion provider, but rather the feature that sets the buffer to an LSP content type.

Reply:
The specific line of code we need to bypass is this:
https://devdiv.visualstudio.com/DevDiv/_git/IntelliCode-VS?path=%2Fsrc%2FVSIX%2FIntelliCode.VSIX%2FRefactorings%2FLanguageServerClient%2FLanguageServerActivator.cs&version=GBmaster&line=59&lineEnd=62&lineStartColumn=1&lineEndColumn=1&lineStyle=plain

Reply:
Actually, I think the above line is irrelevant. The following line will cause the poor behavior regardless of the state of IntelliCode options:
https://devdiv.visualstudio.com/DevDiv/_git/IntelliCode-VS?path=%2Fsrc%2FVSIX%2FIntelliCode.VSIX%2FRefactorings%2FContentTypeDefinitions.cs&version=GBmaster&line=15&lineEnd=20&lineStartColumn=1&lineEndColumn=1&lineStyle=plain

We need to disable the extension itself (as a whole).

Reply:
Agreed, there is currently no way I know of to avoid the content type MEF export without disabling the IntelliCode package.
I am not sure how to do that during test execution.

(@DavidObando fyi)

Reply:
I believe we decided to leave these changes in place until the platform sorts out a way to manage blocking and non-blocking completion sources. Reverting these changes are being tracked by https://github.com/dotnet/roslyn/issues/41968

Reply:
isn't this something that LSP would be responsible for? They are putting platform in non-blocking mode. See `Microsoft.VisualStudio.LanguageServer.Client.AsyncCompletionSourceProvider`

Reply:
Is there an easy way for the LanguageServerClient to have non-blocking completion separate from this option?